### PR TITLE
IGNITE-8841

### PR DESCRIPTION
**Pull request by gvvinblade**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        /** */
+        @GridToStringInclude
+        boolean init;
+
+        /** */
+        @GridToStringInclude
+        Set<Long> cntrs;
+
+        /** */
+        boolean done() {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java]: Would name isDone clearer?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+            if (init && !hasQueries)
+                removeAndCheckDone(nodeId);
+            else
+                node.init = true;
+        }
+    }
+
+    /** */
+    private void removeAndCheckDone(@NotNull UUID nodeId) {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java]: 'check' look misleading here as we actually 'mark' as done, but not make a check.
Would e.g. 'finish(UUID)' clearer?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccAbstractTest.java [hunk 3]
```diff
@@ -1716,12 +1716,19 @@ protected MvccProcessorImpl mvccProcessor(Ignite node) {
      * @throws Exception If failed.
      */
     protected final void checkActiveQueriesCleanup(Ignite node) throws Exception {
-        final MvccProcessorImpl crd = mvccProcessor(node);
+        final MvccProcessorImpl prc = mvccProcessor(node);
```
Comment (AMashenkov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccAbstractTest.java]: Let's rename to 'proc'

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java [hunk 4]
```diff
@@ -3336,6 +3346,102 @@ else if (mappings.single())
...
+    public IgniteInternalFuture<MvccSnapshot> requestSnapshot() {
+        if (isRollbackOnly())
+            return new GridFinishedFuture<>(rollbackException());
+
+        MvccSnapshot mvccSnapshot0 = mvccSnapshot;
+
+        if (mvccSnapshot0 != null)
+            return new GridFinishedFuture<>(mvccSnapshot0);
+
+        MvccProcessor prc = cctx.coordinators();
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java]: Let's rename to 'proc'

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java (hunk 1 above)
Comment (gvvinblade) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java]: possibly, will change

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java (hunk 2 above)
Comment (gvvinblade) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/PreviousQueries.java]: not sure, it is just a couple of doings which is being done in common. It's almost always difficult to come up with right name for such methods. There are two actions here: 1) removing the node from the active list, 2) checking whether the done flag can be set and setting it if it is so. I think the current name is more or less descriptive.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccAbstractTest.java (hunk 3 above)
Comment (gvvinblade) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccAbstractTest.java]: Not sure if it's really needed here. There are too many places where this name is used. We may easily rename all of them but in separate commit.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java (hunk 4 above)
Comment (gvvinblade) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java]: See my comment above

**Pull request closed without merging**